rust/src/storage/file.rs

```diff
@@ -17,6 +18,19 @@ impl FileStorageBackend {
 
 #[async_trait::async_trait]
 impl StorageBackend for FileStorageBackend {
+    fn join_path(
+        &self,
+        path: &str,
+        path_to_join: &str,
+    ) -> String {
+        let new_path = Path::new(path);
+        new_path
+            .join(path_to_join)
+            .into_os_string()
+            .into_string()
+            .unwrap()
```

Comment:
It's hard for me to ever imagine what sequence of events could lead to this `into_string()` call failing, but I will defer to @houqp's judgement on whether `join_path` should propagate its Error

Reply:
i think it should be fairly safe here since we only have `&str` as inputs.